File: ignicao/lib/features/clients/data/usecases/delete_client_usecase.dart
import 'package:dartz/dartz.dart';

import '../../../../shared/data/usecases/usecase.dart';
import '../../../../shared/errors/failures.dart';
import '../../domain/dto/delete_client_dto.dart';
import '../../domain/repositories/client_repository.dart';

class DeleteClientUseCase implements UseCase<void, DeleteClientDto> {
  DeleteClientUseCase(this.repository);

  final ClientRepository repository;

  @override
  Future<Either<Failure, void>> call(DeleteClientDto params) async {
    return repository.deleteClient(params);
  }
}

File: ignicao-api/src/clients/application/usecases/delete-client.usecase.ts
import { ClientRepository } from 'src/clients/domain/repositories/client.repository';
import { UseCase } from 'src/shared/application/usecases/usecase';

export type DeleteInput = {
  id: string;
};

export class DeleteClientUsecase implements UseCase<DeleteInput, void> {
  constructor(private readonly clientRepository: ClientRepository) {}

  async execute(deleteInput: DeleteInput): Promise<void> {
    await this.clientRepository.delete(deleteInput.id);
  }
}
